sycl/source/detail/kernel_compiler/kernel_compiler_sycl.cpp

```diff
@@ -315,3 +315,59 @@ bool SYCL_Compilation_Available() {
 } // namespace _V1
 } // namespace sycl
 #endif
+
+#if SYCL_EXT_JIT_ENABLE
+
+#include "../jit_compiler.hpp"
+
+namespace sycl {
+inline namespace _V1 {
+namespace ext::oneapi::experimental {
+namespace detail {
+
+bool SYCLJIT_Compilation_Available() {
+  return sycl::detail::jit_compiler::get_instance().isAvailable();
+}
+
+spirv_vec_t
+SYCLJIT_to_SPIRV(const std::string &SYCLSource, include_pairs_t IncludePairs,
+                 const std::vector<std::string> &UserArgs, std::string *LogPtr,
+                 const std::vector<std::string> &RegisteredKernelNames) {
+  return sycl::detail::jit_compiler::get_instance().compileSYCL(
+      SYCLSource, IncludePairs, UserArgs, LogPtr, RegisteredKernelNames);
+}
+
+} // namespace detail
+} // namespace ext::oneapi::experimental
+} // namespace _V1
+} // namespace sycl
+
+#else
+
+namespace sycl {
+inline namespace _V1 {
+namespace ext::oneapi::experimental {
+namespace detail {
+
+bool SYCLJIT_Compilation_Available() { return false; }
+
+spirv_vec_t
+SYCLJIT_to_SPIRV(const std::string &SYCLSource, include_pairs_t IncludePairs,
+                 const std::vector<std::string> &UserArgs, std::string *LogPtr,
+                 const std::vector<std::string> &RegisteredKernelNames) {
+  (void)SYCLSource;
+  (void)IncludePairs;
+  (void)UserArgs;
+  (void)LogPtr;
+  (void)RegisteredKernelNames;
+
+  throw sycl::exception(sycl::errc::build,
+                        "kernel_compiler via sycl-jit is not available");
+}
+
```

Comment:
With `[[maybe_unused]]`, we should be able to avoid the double declaration and can just ifdef the body of the function. In any case, we don't need to duplicate the namespace declarations.

Reply:
Yes, that's neat.